lib/rubocop/cop/style/access_modifier_declarations.rb

```diff
@@ -223,11 +223,14 @@ def allow_modifiers_on_alias_method?(node)
           cop_config['AllowModifiersOnAliasMethod'] && access_modifier_with_alias_method?(node)
         end
 
+        # rubocop:disable Metrics/CyclomaticComplexity
         def offense?(node)
           (group_style? && access_modifier_is_inlined?(node) &&
             !node.parent&.if_type? && !right_siblings_same_inline_method?(node)) ||
-            (inline_style? && access_modifier_is_not_inlined?(node))
+            (inline_style? && access_modifier_is_not_inlined?(node) &&
+              select_grouped_def_nodes(node).any?)
         end
+        # rubocop:enable Metrics/CyclomaticComplexity
```

Comment:
Please consider refactoring as shown below due to the complexity of the conditions. This can eliminate the need to add `rubocop:disable` annotation.

```ruby
def offense?(node)
  if group_style?
    return false if node.parent&.if_type?

    access_modifier_is_inlined?(node) && !right_siblings_same_inline_method?(node)
  else
    access_modifier_is_not_inlined?(node) && select_grouped_def_nodes(node).any?
  end
end
```

Reply:
Done 👍🏻